src/test/java/com/wikia/webdriver/common/templates/NewTestTemplate.java

```diff
@@ -43,4 +43,17 @@ public void stop() {
     }
     stopBrowser();
   }
+
+  private String getCountryCode(Method method, Object[] data) {
+    if (method.getAnnotation(GeoEdgeProxy.class) != null) {
+      return method.getAnnotation(GeoEdgeProxy.class).country();
+    } else {
+      for (Object entity : data) {
+        if (entity instanceof GeoCountry) {
+          return entity.toString();
+        }
+      }
+    }
+    return null;
```

Comment:
Rather than null I would recond using an empty string "" , and .isEmpty() check, also - if you anotate a class, it ignores a data provider, is this a desired behaviour?